plugins/english/novelupdates.ts

```diff
@@ -34,38 +34,41 @@ class NovelUpdates implements Plugin.PluginBase {
         return novels;
     }
 
-    async popularNovels(page: number, { showLatestNovels, filters }: Plugin.PopularNovelsOptions<typeof this.filters>): Promise<Plugin.NovelItem[]> {
-        let link = `${this.site}`;
-        if (filters.language.value.length ||
-            filters.novelType.value.length ||
-            filters.genres.value.include?.length ||
-            filters.genres.value.exclude?.length ||
-            filters.storyStatus.value !== '' ){
-        link += "series-finder/?sf=1"
-        } else if (showLatestNovels){
-            link += "latest-series/?st=1" 
-            } else {
-            link += "series-ranking/?rank=week"
-            };
-        
-        if (filters.language.value.length)
-            link += '&org=' + filters.language.value.join(',');
-        
-        if (filters.novelType.value.length)
-            link += '&nt=' + filters.novelType.value.join(',');
-
-        if (filters.genres.value.include?.length)
-            link += '&gi=' + filters.genres.value.include.join(',');
-
-        if (filters.genres.value.exclude?.length)
-            link += '&ge=' + filters.genres.value.exclude.join(',');
-
-        if (filters.genres.value.include?.length || filters.genres.value.exclude?.length)
-            link += '&mgi=' + filters.genre_operator.value;
-
-        if (filters.storyStatus.value.length)
-            link += '&ss=' + filters.storyStatus.value;
+    async popularNovels(
+        page: number,
+        { showLatestNovels, filters }: Plugin.PopularNovelsOptions<typeof this.filters>
+    ): Promise<Plugin.NovelItem[]> { 
+        let link = this.site;
+
+        const filterConditions = [
+            { check: filters.language.value.length > 0, param: 'org', value: filters.language.value },
+            { check: filters.novelType.value.length > 0, param: 'nt', value: filters.novelType.value },
+            { check: (filters.genres.value.include?.length || 0) > 0, param: 'gi', value: filters.genres.value.include },
+            { check: (filters.genres.value.include?.length || 0) > 0, param: 'mgi', value: filters.genre_operator.value },
```

Comment:
why the `check` uses `filters.geners` but the `value` uses `filters.genre_operator`

Reply:
after testing NU series finder I noticed that if only the `include` part of the array is true, we append the `genre_operator`. It doesn't apply to the `exclude` array